hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieRecordPayload.java

```diff
@@ -56,4 +57,10 @@
      * Return EMPTY to skip writing this record.
      */
     Optional<IndexedRecord> getInsertValue(Schema schema) throws IOException;
+
+    /**
+     * This method can be used to fetch metadata regarding the HoodieRecordPayload, for the purpose
+     * of capturing some metrics.
+     */
+    Map<String, String> getMetadata();
```

Comment:
I am a little concerned with having a map of properties on the record payload. Adding data to record payload is very tricky in terms of shuffle. It put in a 100 properties and tuned it for 10 - things may fail miserably during a shuffle. If we need to encode data - why dont we just do it in the InsertRecord. Can you may be give me an example where storing it in the record itself would not make sense?

Reply:
CMIIW, even if we encode it in the InsertRecord (or HoodieRecordPayload), we need this method to fetch the metadata just before HoodieRecordPayload is deflated like in https://github.com/uber/hoodie/pull/251/commits/8f2c88b836b25d5bd66859abcc2dd49d07bb3e28#diff-165908ecbfe7dbae538c00e8157e7be0R129

Regarding storing properties in a map - did you recommend storing it in the IndexedRecord instead? My concern was inadvertent stuff being put into IndexRecord affecting parquet schema. But we are planning to do that anyway for across_batch_de-duped counts, so yes, I could actually move these properties into IndexedRecord itself.

Reply:
IIUC you want a mechanism to pass some data from each record and have it propagated down to a custom WriteStatus class? Since it was mostly you and @prazanna who discussed this and neither the ticket nor the PR describes what the commit is about, @kaushikd49 could you please summarize what the real use-case is ?

Reply:
I am okay with this approach @kaushikd49. We may want to enable adding properties to HoodieRecordPayload with a property in HoodieWriteConfig (disabled by default). It may be good idea to write up a small doc on how you are planning to use this map downstream internally and share it with @vinothchandar - just so he has context. Thanks and sorry for the delay

Reply:
@vinothchandar Yea that is right. I will be sharing a doc with you shortly explaining the use-case.

@prazanna Sure, I'll do that.